tests/test_noninteractive_mode.py

```diff
@@ -0,0 +1,49 @@
+""" Non-interactive tests. """
+import subprocess
+from mssqltestutils import create_mssql_cli
+
+def test_session_closure_query_valid():
+    """ Test session closure for valid query. """
+    assert is_processed_closed("select 1")
+
+def test_session_closure_query_invalid():
+    """ Test session closure for invalid query. """
+    assert is_processed_closed("asdfasoifjas")
+
+def test_query_large():
+    """ Test against query with over 1000 lines """
+    query_str = "SELECT * FROM STRING_SPLIT(REPLICATE(CAST('X,' AS VARCHAR(MAX)), 1024), ',')"
+    p = subprocess.Popen("./mssql-cli -Q \"%s\"" % query_str, shell=True, stdin=subprocess.PIPE,
+                         stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = p.communicate()[0].decode("utf-8")
+    output_baseline = get_file_contents('./tests/query_tests/big_query.txt')
+    assert output == output_baseline
+
+def test_query_output():
+    """ Tests if -Q has equal output to execute_query. """
+    query_str = "select 1"
+    p = subprocess.Popen("./mssql-cli -Q '%s'" % query_str, shell=True, stdin=subprocess.PIPE,
+                         stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = p.communicate()[0].decode("utf-8").rstrip('\n')
+    output_baseline = get_file_contents('./tests/query_tests/small_query.txt')
+    assert output == output_baseline
+
+def is_processed_closed(query_str):
+    """ Runs unit tests on process closure given a query string """
+    mssql_cli = create_mssql_cli()
+    mssql_cli.execute_query(query_str)
+    mssql_cli.shutdown()
```

Comment:
I wouldn't expect this method to execute a query given it's name. Consider renaming?

Reply:
Please feel free to leave suggestions :) what do you think is better than `execute_query()`?